Source/WebGPU/WGSL/ConstantFunctions.h

```diff
@@ -165,31 +165,31 @@ static ConstantResult constantUnaryOperation(const FixedVector<ConstantValue>& a
     return scalarOrVector([&](auto& arg) -> ConstantResult {
         if constexpr (constraint & Constraints::Bool) {
             if (auto* boolean = std::get_if<bool>(&arg))
-                return { { fn(*boolean) } };
+                return { ConstantValue { fn(*boolean) } };
```

Comment:
The original code is a bit hacky, but the reason this won't work is that `fn(...)` might return a primitive value, a `ConstantValue` or a `ConstantResult`, so you can't always construct a `ConstantValue` from it.

Reply:
I've tried it again. Without `ConstantValue` there GCC complains about ambiguous resolution of the ConstantResult (a.k.a. `WTF::Expected<ConstantValue,String>`) constructor.

Attaching 
[error.txt](https://github.com/user-attachments/files/17780762/error.txt)

but the gist of it is:
```
/project/webkit/WebKit/Source/WebGPU/WGSL/ConstantFunctions.h:168:43: error: conversion from ‘<brace-enclosed initializer list>’ to ‘WGSL::ConstantResult’ {aka ‘std::experimental::fundamentals_v3::expected<WGSL::ConstantValue, WTF::String>’} is ambiguous
  168 |                 return { { fn(*boolean) } };
      |                                           ^
In file included from /project/webkit/WebKit/build.release/WTF/Headers/wtf/text/StringImpl.h:35,
                 from /project/webkit/WebKit/build.release/WTF/Headers/wtf/text/WTFString.h:32,
                 from /project/webkit/WebKit/Source/WebGPU/WGSL/CompilationMessage.h:30,
                 from /project/webkit/WebKit/Source/WebGPU/WGSL/WGSL.h:28,
                 from /project/webkit/WebKit/Source/WebGPU/WGSL/TypeCheck.h:28,
                 from /project/webkit/WebKit/Source/WebGPU/WGSL/TypeCheck.cpp:27:
/project/webkit/WebKit/build.release/WTF/Headers/wtf/Expected.h:301:15: note: candidate: ‘constexpr std::experimental::fundamentals_v3::expected< <template-parameter-1-1>, <template-parameter-1-2> >::expected(unexpected_type&&) [with T = WGSL::ConstantValue; E = WTF::String; unexpected_type = std::experimental::fundamentals_v3::unexpected<WTF::String>]’
  301 |     constexpr expected(unexpected_type&& u) : base(__expected_detail::error_tag, std::forward<unexpected_type>(u).value()) { }
      |               ^~~~~~~~
/project/webkit/WebKit/build.release/WTF/Headers/wtf/Expected.h:300:15: note: candidate: ‘constexpr std::experimental::fundamentals_v3::expected< <template-parameter-1-1>, <template-parameter-1-2> >::expected(const unexpected_type&) [with T = WGSL::ConstantValue; E = WTF::String; unexpected_type = std::experimental::fundamentals_v3::unexpected<WTF::String>]’
  300 |     constexpr expected(const unexpected_type& u) : base(__expected_detail::error_tag, u.value()) { }
      |               ^~~~~~~~
/project/webkit/WebKit/build.release/WTF/Headers/wtf/Expected.h:297:15: note: candidate: ‘constexpr std::experimental::fundamentals_v3::expected< <template-parameter-1-1>, <template-parameter-1-2> >::expected(value_type&&) [with T = WGSL::ConstantValue; E = WTF::String; value_type = WGSL::ConstantValue]’
  297 |     constexpr expected(value_type&& e) : base(__expected_detail::value_tag, std::forward<value_type>(e)) { }
      |               ^~~~~~~~
/project/webkit/WebKit/build.release/WTF/Headers/wtf/Expected.h:296:15: note: candidate: ‘constexpr std::experimental::fundamentals_v3::expected< <template-parameter-1-1>, <template-parameter-1-2> >::expected(const value_type&) [with T = WGSL::ConstantValue; E = WTF::String; value_type = WGSL::ConstantValue]’
  296 |     constexpr expected(const value_type& e) : base(__expected_detail::value_tag, e) { }
```

I'm continuing to look but it might be something like this GCC bug. (but it's not clear if it's a bug in GCC).
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=59389#c8 

I'm not 100% clear on it but it might be a confusion about interpreting the inner pair of braces as constructing a ConstantResult and then the outer pair as invoking ConstantResult's ordinary copy or move constructor.

I'll try Clang.

Reply:
FYI: I'm using GCC 13.2.0

Reply:
Looks like Clang on Linux handles the original code just fine.  I'm happy about that, and will only claim Clang works.

Reply:
Removed these changes, and only claim Clang works.  I am using Clang 16.0.6